Source/TrustEvaluator.swift

```diff
@@ -87,14 +91,15 @@ final internal class TrustEvaluator: TrustEvaluatorType {
     
     private func evaluate(_ trust: SecTrust) -> URLSession.AuthChallengeDisposition {
         let count = SecTrustGetCertificateCount(trust)
-        let trustCertificates = [Int](0..<count).compactMap { SecTrustGetCertificateAtIndex(trust, $0) }
-        let serverPinnables = strategy.pinnables(from: trustCertificates)
+        let serverCerts = [Int](0..<count).compactMap { SecTrustGetCertificateAtIndex(trust, $0) }
+        var serverKeys = [SecKey]()
+        if #available(iOS 10.3, *) {
+            serverKeys = serverCerts.compactMap { SecCertificateCopyPublicKey($0) }
+        }
         
-        for serverPinnable in serverPinnables.reversed() {
-            for pinnable in pinnables {
-                if pinnable.isEqualTo(serverPinnable) {
-                    return .performDefaultHandling
-                }
+        for serverKey in serverKeys.reversed() {
```

Comment:
Why `.reversed()`?

Reply:
Because the certificate chain in the auth challenge is sorted such that the root CA is at the end. Since we're pinning against that it makes sense to check it first. Just a small optimization, the match *should* happen on the first check rather than the ninth.